convert2rhel/checks.py

```diff
@@ -95,56 +97,59 @@ def check_convert2rhel_latest():
     utils.mkdir_p(repo_dir)
     repo_path = os.path.join(repo_dir, "convert2rhel.repo")
-    store_content_to_file(filename=repo_path, content=repo_content)
+    store_content_to_file(filename=repo_path, content=REPO_CONTENT)
 
     cmd.append("--setopt=reposdir=%s" % repo_dir)
     cmd.append("convert2rhel")
 
     raw_output_convert2rhel_versions, return_code = run_subprocess(cmd, print_output=False)
 
-    if return_code != 0:
-        logger.warning(
-            "Couldn't check if this is the latest version of convert2rhel\n"
-            "repoquery failed %s, %s" % (return_code, raw_output_convert2rhel_versions)
-        )
-        return
-
     PKG_NEVR = r"\b(\S+)-(?:([0-9]+):)?(\S+)-(\S+)\b"
     convert2rhel_versions = re.findall(PKG_NEVR, raw_output_convert2rhel_versions, re.MULTILINE)
-    latest_version = ("0", "0.00", "0")
+    latest_available_version = ("0", "0.00", "0")
 
+    # This loop will determin the latest_available_version in the convert2rhel yum repo
+    # it assigns the epoch ,version, and release ex:("0" "0.26" "1.el7") to the latest_available_version var.
     for package_version in convert2rhel_versions:
-        # rpm.lableCompare(pkg1, pkg2) compare two kernel version strings and return
-        # -1 if str2 is greater then str1, 0 if they are equal, 1 if str1 is greater the str2
-        ver_compare = rpm.labelCompare(package_version[1:], latest_version)
+        # rpm.lableCompare(pkg1, pkg2) compare two package version strings and return
+        # -1 if lastest_version is greater then package_version, 0 if they are equal, 1 if package_version is greater the lastest_version
+        ver_compare = rpm.labelCompare(package_version[1:], latest_available_version)
         if ver_compare > 0:
-            latest_version = package_version[1:]
-
-    # After the for loop latest_version will have the epoch ,version, and release ex:("0" "0.26" "1.el7") information from convert2rhel yum repo.
-    # the release for latest_verion will be "1.el7" and the relase for convert2rhel_version will be hard coded as "0" below,
-    # therefore when the versions are the same the latest_version's release field will cause it to evaluate as later
-    ver_compare = rpm.labelCompare(("0", convert2rhel_version, "0"), ("0", latest_version[1], "0"))
+            latest_available_version = package_version[1:]
+
+    # we hard code 0 for the epoch and release below because latest_verion will
+    # contain "1.el7" for the release but we don't know what the release for
+    # installed_convert2rhel_version will be, if we did not hard code to 0 latest_available_version
+    # will evaluate as later then installed_convert2rhel_version even when the versions
+    # were the same.
+    #
+    # ver_compare is set to
+    # * -1 if lastest_version is greater then installed_convert2rhel_version,
+    # * 0 if they are equal
+    # * 1 if installed_convert2rhel_version is greater the lastest_version
```

Comment:
Do we need this comment since we have a more generic one in line 115? I mean, this might help to identify what is the expected output of the `ver_compare` variable, but I would say it's redundant since we have the same information on line 115.

Reply:
I agree that I shouldn't repeat this comment as it is redundant. Thank you!